sydent/http/servlets/versions.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2022 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from twisted.web.server import Request
+
+from sydent.http.servlets import SydentResource, jsonwrap, send_cors
+from sydent.types import JsonDict
+
+
+class VersionsServlet(SydentResource):
+    isLeaf = True
+
+    @jsonwrap
+    def render_GET(self, request: Request) -> JsonDict:
+        """
+        Get the terms that must be agreed to in order to use this service
+        Returns: Object describing the terms that require agreement
+        """
+        send_cors(request)
+
+        return {"versions": ["r0.1.0", "r0.2.0", "r0.2.1", "r0.3.0", "v1.1"]}
```

Comment:
As far as I could tell the versions just added more features, so I *think* this is true.

@turt2live should this include `v1.0` since that had no changes from `r0.3.0` besides the global versioning?

Reply:
`v1.0` is not a real Matrix version, so no for that one. 

If there's no changes for identity servers, including the extra versions is fine though theoretically inconsequential. I'd probably do it while we're here though.

Reply:
The other r0 versions will be affected when the old `/api/v1` endpoints get dropped in https://github.com/matrix-org/sydent/issues/338 btw

Reply:
> If there's no changes for identity servers, including the extra versions is fine though theoretically inconsequential. I'd probably do it while we're here though.

Makes sense to me. 👍 Although there's no changes I could see someone reading the `v1.5` spec and then wondering why sydent doesn't support it, when in actuality it does.

> The other r0 versions will be affected when the old `/api/v1` endpoints get dropped in #338

Yep, but they're supported now so. 🤷